sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -159,37 +160,17 @@ public HttpTransport(
     int responseCode = -1;
 
     try (final OutputStream outputStream = connection.getOutputStream();
-        final Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream, UTF_8))) {
+        final GZIPOutputStream gzip = new GZIPOutputStream(outputStream);
+        final Writer writer = new BufferedWriter(new OutputStreamWriter(gzip, UTF_8))) {
       serializer.serialize(event, writer);
 
-      // need to also close the input stream of the connection
-      connection.getInputStream().close();
+      finishAndFlushStreams(outputStream, gzip);
```

Comment:
If I understood "try-with-resources", it will call `outputStream.close()` when the try block is over, and `close` calls `flush` according to https://docs.oracle.com/javase/8/docs/api/index.html?java/io/OutputStreamWriter.html.

Do we need to flush it in `finishAndFlushStreams`?!

Similarly, `GZIPOutputStream` inherits `close` from https://docs.oracle.com/javase/8/docs/api/index.html?java/io/OutputStreamWriter.html, which is documented to have the same effect as `finish` + closing the underlying stream (which is again our `outputStream`).

Reply:
@rhcarvalho you're right, but https://developer.android.com/reference/android/os/StrictMode for some reason complains that there's something open and it should be closed, also a few issues on SO and people said to ignore it, but I prefer to `flush` it again for the sake of having a clean log for our customers.

Reply:
![Screenshot 2020-05-27 at 12 21 18](https://user-images.githubusercontent.com/5731772/83007926-abc7a780-a014-11ea-8c30-2ef6d2981239.png)

Reply:
Hmm the log message talks about a _response body_, and what we are finishing and flushing here are streams related to the **request** body, no?!

Reply:
Where is the response body being closed?

Reply:
@rhcarvalho this warning was noticed before this PR and the response stream was always closed:
https://github.com/getsentry/sentry-android/blob/master/sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java#L166

I changed again, let's see if I can experience the log again or not, it's intermittent, but when debugging you can check the status of the streams and they are all closed at least.

Reply:
👍  Will let Bruno have his final look.